### PR TITLE
Type-erase `DoAll` eventuals for generated code

### PR DESCRIPTION
**Pull request by onelxj**

### CODE REVIEW

File: protoc-gen-eventuals/templates/eventuals.cc.j2 [hunk 1]
```diff
@@ -27,11 +27,10 @@ using eventuals::grpc::Stream;
 using namespace {{ namespaces | join('::') }}::eventuals;
 
 {% for service in services -%}
-Task::Of<void> {{ service.name }}::TypeErasedService::Serve() {
-  return [this]() {
-    return DoAll(
 {%- for method in service.methods %}
-{%- set input_type -%}
+Task::Of<void> {{ service.name }}::TypeErasedService::{{ method.name }}Extractor() {
```
Comment (benh) [inline protoc-gen-eventuals/templates/eventuals.cc.j2:31]:
```suggestion
Task::Of<void> {{ service.name }}::TypeErasedService::Serve{{ method.name }}() {
```

File: protoc-gen-eventuals/templates/eventuals.cc.j2 [hunk 2]
```diff
@@ -125,8 +123,17 @@ Task::Of<void> {{ service.name }}::TypeErasedService::Serve() {
...
+        };
+}
+{% endfor %}
+
+Task::Of<void> {{ service.name }}::TypeErasedService::Serve() {
+  return [this]() {
+    return DoAll(
+{%- for method in service.methods %}
+      // {{ method.name }}
+      {{ method.name }}Extractor(){%- if not loop.last -%},{%- endif %}
```
Comment (benh) [inline protoc-gen-eventuals/templates/eventuals.cc.j2:136]:
```suggestion
      Serve{{ method.name }}(){%- if not loop.last -%},{%- endif %}
```

File: protoc-gen-eventuals/templates/eventuals.cc.j2 [hunk 3]
```diff
@@ -125,8 +123,17 @@ Task::Of<void> {{ service.name }}::TypeErasedService::Serve() {
...
+          | Loop();
+        };
+}
+{% endfor %}
+
+Task::Of<void> {{ service.name }}::TypeErasedService::Serve() {
+  return [this]() {
+    return DoAll(
+{%- for method in service.methods %}
+      // {{ method.name }}
```
Comment (benh) [inline protoc-gen-eventuals/templates/eventuals.cc.j2:135]: Can delete this now:
```suggestion
```

File: protoc-gen-eventuals/templates/eventuals.h.j2 [hunk 4]
```diff
@@ -52,6 +52,11 @@ struct {{ service.name }} {
             ::grpc::GenericServerContext*,
             {{ input_type }}*>* args) = 0;
 {% endfor %}
+
+    private:
+{% for method in service.methods %}
+    ::eventuals::Task::Of<void> {{ method.name }}Extractor();
```
Comment (benh) [inline protoc-gen-eventuals/templates/eventuals.h.j2:58]:
```suggestion
    ::eventuals::Task::Of<void> Serve{{ method.name }}();
```

**Review by benh — APPROVED** [review]

**Review by ArthurBandaryk — APPROVED** [review]

**Pull request merged**